libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1545,7 +1581,7 @@ string YulUtilFunctions::copyArrayToStorageFunction(ArrayType const& _fromType,
 		templ("arrayLength",arrayLengthFunction(_fromType));
 		templ("isValueType", _fromType.baseType()->isValueType());
 		templ("dstDataLocation", arrayDataAreaFunction(_toType));
-		if (!fromCalldata || _fromType.baseType()->isValueType())
+		if (fromMemory || _fromType.baseType()->isValueType())
```

Comment:
I think this should be changed to:
```suggestion
		if (fromMemory || (fromCalldata && _fromType.baseType()->isValueType()))
```